File: packages/python-sdk/src/schemax/providers/unity/hierarchy.py
"""
Unity Catalog Hierarchy Configuration
"""

from schemax.providers.base.hierarchy import Hierarchy, HierarchyLevel
from schemax.providers.base.hierarchy_defaults import build_table_level

# Unity Catalog has a 3-level hierarchy: Catalog → Schema → Table
unity_hierarchy_levels = [
    HierarchyLevel(
        name="catalog",
        display_name="Catalog",
        plural_name="catalogs",
        icon="database",
        is_container=True,
    ),
    HierarchyLevel(
        name="schema",
        display_name="Schema",
        plural_name="schemas",
        icon="folder",
        is_container=True,
    ),
    build_table_level(),
]

unity_hierarchy = Hierarchy(unity_hierarchy_levels)

File: packages/vscode-extension/src/providers/unity/hierarchy.ts
/**
 * Unity Catalog Hierarchy Configuration
 */

import type { HierarchyLevel } from '../base/hierarchy';
import { Hierarchy } from '../base/hierarchy';

/**
 * Unity Catalog has a 3-level hierarchy:
 * Catalog → Schema → Table (and other objects)
 */
export const unityHierarchyLevels: HierarchyLevel[] = [
  {
    name: 'catalog',
    displayName: 'Catalog',
    pluralName: 'catalogs',
    icon: 'database',
    isContainer: true,
  },
  {
    name: 'schema',
    displayName: 'Schema',
    pluralName: 'schemas',
    icon: 'folder',
    isContainer: true,
  },
  {
    name: 'table',
    displayName: 'Table',
    pluralName: 'tables',
    icon: 'table',
    isContainer: false,
  },
];

export const unityHierarchy = new Hierarchy(unityHierarchyLevels);

